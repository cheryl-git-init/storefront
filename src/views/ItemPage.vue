<template>
  <div>
    <NavBar />
    <div class="item">
      <h2>{{ getItemInDepth.name }}</h2>
      <img v-bind:src="getImgLink()" height="100px" />
      <p>{{ getItemInDepth.currencySymbol }}{{ (getItemInDepth.usdPrice/100).toFixed(2) }}</p>
      <AddToBasket :id="id" />
      <p>{{ getItemInDepth.description }}</p>
    </div>
  </div>
</template>

<script>
// @ is an alias to /src
import NavBar from "@/components/NavBar.vue";
import AddToBasket from "@/components/AddToBasket.vue";

export default {
  name: "ItemPage",
  components: {
    NavBar,
    AddToBasket
  },
  props: {
    id: String
  },
  computed: {
    getItemInDepth() {
      return this.$store.getters.getItemInDepth(this.id);
    }
  },
  mounted: function() {
    this.$store.dispatch("getAllItems");
  },
  methods: {
    getImgLink() {
      return require("@/assets/" + this.id + "_t.png");
    }
  }
};
</script>

<style lang="scss" scoped>
.item {
  top: 70px;
  position: relative;
}
</style>