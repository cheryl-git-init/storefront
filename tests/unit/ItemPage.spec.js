import Vuex from 'vuex';
import ItemPage from '@/views/ItemPage.vue';
import { shallowMount, createLocalVue } from '@vue/test-utils';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('ItemPage.vue', () => {
    const actions = {
        getAllItems: jest.fn()
    }
    const store = new Vuex.Store({
        actions
    })
    
    const wrapper = shallowMount(ItemPage, {
        namespaced: false,
        computed: { getItemInDepth() { return { name: "test", currencySymbol: "$", usdPrice: 200, description: "test here" } } },
        methods: { getImgLink() {return ""} },
        store,
        localVue,
    })
    it('recieves and displays the correct data', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })
    it('calls getAllItems on load', () => {
        expect(actions.getAllItems).toBeCalled()
    })
})
