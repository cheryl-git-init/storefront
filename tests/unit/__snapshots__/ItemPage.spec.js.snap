// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemPage.vue recieves and displays the correct data 1`] = `
<div>
  <navbar-stub></navbar-stub>
  <div class="item">
    <h2>test</h2> <img src="" height="100px">
    <p>$2.00</p>
    <addtobasket-stub></addtobasket-stub>
    <p>test here</p>
  </div>
</div>
`;
